driver-core/src/main/com/mongodb/client/model/expressions/MqlExpression.java

```diff
@@ -161,6 +170,72 @@ public BooleanExpression lte(final Expression lte) {
         return new MqlExpression<>(ast("$lte", lte));
     }
 
+    public BooleanExpression isBoolean() {
```

Comment:
https://github.com/mongodb/mongo-java-driver/pull/1032/files#r1032636669. I will do this at the end, sometimes they are public because they are used in a future PR.